newdle/api.py

```diff
@@ -77,6 +81,38 @@ def require_token():
     g.user = user
 
 
+@api.route('/avatar/<email>')
+@allow_anonymous
+def user_avatar(email):
+    size = request.args.get('size')
+    email_hex = hashlib.md5(email.encode()).hexdigest()
+    # make gravat return 404 HTTP code instead of a default image
+    query_args = {'d': '404'}
+    if size is not None:
+        query_args['s'] = size
+    gravatar_url = (
+        f'https://gravatar.com/avatar/{quote(email_hex)}?' f'{urlencode(query_args)}'
+    )
+    resp = requests.get(gravatar_url)
+    if resp.status_code == 404:
+        user = search_users(None, email, 1)[1][0]
+        return render_user_avatar(user, size)
+
+    # filter out headers that might break the response
+    excluded_headers = [
+        'content-encoding',
+        'content-length',
+        'transfer-encoding',
+        'connection',
+    ]
+    headers = [
+        (name, value)
+        for (name, value) in resp.raw.headers.items()
+        if name.lower() not in excluded_headers
+    ]
```

Comment:
I guess this is needed to handle last-modified-since etc. correctly? Otherwise I'd just use a whitelist and only forward relevant headers (e.g. content-type)

Reply:
This is also to prevent a case in which we get `transfer-encoding` == gzip. Flask decompresses the response body for us, so we either have to gzip the response again or get rid of the headers that might cause some trouble.

Reply:
The question was more about why we need to forward response headers in the first place, not why we filter some out ;)

Reply:
Ok then. Let's then agree on some necessary headers we might want to forward. 'Content-Type', 'Last-Modified', 'Cache-Control', 'Expires', 'Date', 'Content-Disposition'?

Reply:
etag (or e-tag?) as well